src/main/scala/ie/zalando/fabric/gateway/service/IngressDerivationChain.scala

```diff
@@ -90,8 +90,33 @@ class IngressDerivationChain(stackSetOperations: StackSetOperations)(implicit ma
       val finalRoutes = gateway.corsConfig.fold(skipperRoutes) { corsConfig =>
         withCors(gateway, meta.name, corsConfig, skipperRoutes)
       }
-      combineBackendsAndRoutes(backends, finalRoutes, meta)
+      val ingressDefinitions = combineBackendsAndRoutes(backends, finalRoutes, meta)
+
+      gateway.serviceProvider match {
+        case StackSetProvidedServices(_, _) if versionedHostsEnabled => appendServiceSpecificHosts(ingressDefinitions)
+        case _ => ingressDefinitions
+      }
+    }
+  }
+
+  def appendServiceSpecificHosts(ingresses: List[IngressDefinition]): List[IngressDefinition] = {
```

Comment:
Need to use a ubiquitous language for this feature. Can this be renamed to something like:
`appendVersionedHosts`

Same comment everywhere else you see split terminology been introduced